modules/core/src/main/java/org/apache/ignite/internal/binary/BinaryEnumObjectImpl.java

```diff
@@ -174,10 +174,40 @@ public BinaryEnumObjectImpl(BinaryContext ctx, byte[] arr) {
     }
 
     /** {@inheritDoc} */
-    @Override public <T> T deserialize() throws BinaryObjectException {
-        Class cls = BinaryUtils.resolveClass(ctx, typeId, clsName, ctx.configuration().getClassLoader(), false);
+    @Override public <T> T deserialize(@Nullable ClassLoader ldr) throws BinaryObjectException {
+        ClassLoader resolveLdr = ldr == null ? ctx.configuration().getClassLoader() : ldr;
+
+        Class cls = BinaryUtils.resolveClass(ctx, typeId, clsName, resolveLdr, false);
+
+        return (T)(ldr == null ? BinaryEnumCache.get(cls, ord) : uncachedValue(cls));
+    }
+
+    /**
+     * Get value for the given class without any caching.
```

Comment:
Add a description of parameter ord.

Reply:
Seems like it's not actual: there's no parameter "ord"